### PR TITLE
respond to the case where the class name is written in lower case

### PR DESCRIPTION
**Pull request by nodai2hITC**
I'm thinking of adding suggestion when class name is written in lower case.
```ruby
current_time = time.now
# NameError (undefined local variable or method `time' for main:Object)
# Did you mean?  Time
```

### CODE REVIEW

**Comment by yuki24** [comment]
I'm slightly concerned about too many or inaccurate suggestions. These days, DYM suffers more from not being accurate (see example below) than not suggesting possible names.

```ruby
require 'pathname'
paths
# => NameError (undefined local variable or method `paths' for main:Object)
#    Did you mean?  Pathname
```

Do you have any statstically significant analysis for after/before this change?

File: lib/did_you_mean/spell_checkers/name_error_checkers/class_names.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# frozen-string-literal: true
+require 'delegate'
+
+module DidYouMean
+  module ClassNames
```
Comment (yuki24) [inline lib/did_you_mean/spell_checkers/name_error_checkers/class_names.rb]: I'm not sure if this is a good name - I would probably just change the `DidYouMean::VariableNameChecker` to depend on the `DidYouMean::ClassNameChecker` for class name suggestions. Something along the lines of:

```diff
--- a/lib/did_you_mean/spell_checkers/name_error_checkers/variable_name_checker.rb
+++ b/lib/did_you_mean/spell_checkers/name_error_checkers/variable_name_checker.rb
@@ -2,9 +2,11 @@

 require "did_you_mean/spell_checker"

+require_relative './class_name_checker'
+
 module DidYouMean
   class VariableNameChecker
-    attr_reader :name, :method_names, :lvar_names, :ivar_names, :cvar_names
+    attr_reader :name, :method_names, :lvar_names, :ivar_names, :cvar_names, :class_name_checker

     NAMES_TO_EXCLUDE = { 'foo' => [:fork, :for] }
     NAMES_TO_EXCLUDE.default = []
@@ -71,12 +73,15 @@ module DidYouMean
       @ivar_names   = receiver.instance_variables
       @cvar_names   = receiver.class.class_variables
       @cvar_names  += receiver.class_variables if receiver.kind_of?(Module)
+
+      @class_name_checker = ClasssNameChecker.new(exception)
     end

     def corrections
       @corrections ||= SpellChecker
                      .new(dictionary: (RB_RESERVED_WORDS + lvar_names + method_names + ivar_names + cvar_names))
                      .correct(name) - NAMES_TO_EXCLUDE[@name]
+                     + class_name_checker.corrections
     end
   end
 end
```

(of course this is untested)

**Comment by nodai2hITC** [comment]
> I'm slightly concerned about too many or inaccurate suggestions. These days, DYM suffers more from not being accurate (see example below) than not suggesting possible names.

So, my idea only suggests if the spelling is the same but the case is different.

> Do you have any statstically significant analysis for after/before this change?

I have no "statstically significant analysis". But I have teached ruby programming to many beginners, I often see this kind of mistake.

> I'm not sure if this is a good name - I would probably just change the DidYouMean::VariableNameChecker to depend on the DidYouMean::ClassNameChecker for class name suggestions. 

Thanks your opinion. I tried to rewrite in the way.

**Comment by yuki24** [comment]
I'm closing this issue. While I like the idea of making Ruby more friendly to beginners, I don't think this is something we want to enable globally. Perhaps something like a beginner mode, where more educational suggestions, might be more appropriate. However, that seems like a different type of effort we should make outside of this project.

**Pull request closed without merging**